### PR TITLE
Adding the first snippet for the Logging API

### PR DESCRIPTION
**Pull request by tmatsuo**

### CODE REVIEW

File: logging/src/GoogleProjectTrait.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Samples\Logging;
+
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+/**
+ * Trait for determining the current project ID using "gcloud"
```
Comment (SurferJeffAtGoogle) [inline logging/src/GoogleProjectTrait.php]: Add a note that the class using the trait must inherit from Symfony\Component\Console\Command\Command.

File: logging/src/GoogleProjectTrait.php [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            'The Google Cloud Platform project name to use for this command. ' .
+            'If omitted then the current gcloud project is assumed. ',
+            $this->getProjectIdFromGcloud()
+        );
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function interact(InputInterface $input)
```
Comment (SurferJeffAtGoogle) [inline logging/src/GoogleProjectTrait.php]: What if multiple traits implement interact?

File: logging/src/GoogleProjectTrait.php (hunk 1 above)
Comment (tmatsuo) [inline logging/src/GoogleProjectTrait.php]: I started to think I should make this a subclass of `Symfony\Component\Console\Command\Command`. I'll change it so in the next push.

File: logging/src/GoogleProjectTrait.php [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+            null,
+            InputOption::VALUE_REQUIRED,
+            'The Google Cloud Platform project name to use for this command. ' .
+            'If omitted then the current gcloud project is assumed. ',
+            $this->getProjectIdFromGcloud()
+        );
+    }
+
+    /**
+     * @inheritdoc
```
Comment (tmatsuo) [inline logging/src/GoogleProjectTrait.php]: Good point, I'll subclass so it'll make sense

File: logging/src/GoogleProjectTrait.php (hunk 2 above)
Comment (tmatsuo) [inline logging/src/GoogleProjectTrait.php]: http://php.net/manual/en/language.oop5.traits.php
You have to resolve conflict manually, or it'll throw an error

**Comment by tmatsuo** [comment]
@SurferJeffAtGoogle Thanks! PTAL

**Comment by SurferJeffAtGoogle** [comment]
LGTM.

**Comment by tmatsuo** [comment]
Thanks!

File: logging/logging-sample-cli.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env php
```
Comment (bshaffer) [inline logging/logging-sample-cli.php]: We've been using underscores for PHP filesnames.

File: logging/src/CommandWithProject.php [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+namespace Google\Cloud\Samples\Logging;
+
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+/**
+ * A base class for commands which needs project id.
+ */
+class CommandWithProject extends Command
```
Comment (bshaffer) [inline logging/src/CommandWithProject.php]: This should be abstract

File: logging/src/Write.php [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+            );
+        $this->addProjectOption();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $message = $input->getArgument("message");
+        $project = $input->getOption('project');
+        // [START write_log]
+        $logging = new LoggingClient(['projectId' => $project]);
```
Comment (bshaffer) [inline logging/src/Write.php]: This should be `$projectId` to be consistent with the rest of the samples.

File: logging/src/Write.php [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+                InputArgument::OPTIONAL,
+                "The log message to write",
+                "Hello"
+            );
+        $this->addProjectOption();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $message = $input->getArgument("message");
```
Comment (bshaffer) [inline logging/src/Write.php]: This should be single quotes.

File: logging/src/CommandWithProject.php [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+        exec(
+            "gcloud config list --format 'value(core.project)' 2>/dev/null",
+            $output,
+            $return_var
+        );
+        if (0 === $return_var) {
+            return array_pop($output);
+        }
+        return null;
+    }
```
Comment (bshaffer) [inline logging/src/CommandWithProject.php]: did we actually abstract an entire class for a single command?

This adds indirection and abstraction without even the advantage of DRY. Although even w/ DRY it's not worth it. For samples especially, gaining clarity is worth repeating code.

File: logging/src/Write.php (hunk 7 above)
Comment (tmatsuo) [inline logging/src/Write.php]: will fix in the next PR

File: logging/src/Write.php (hunk 6 above)
Comment (tmatsuo) [inline logging/src/Write.php]: will fix in the next PR

File: logging/src/CommandWithProject.php (hunk 8 above)
Comment (tmatsuo) [inline logging/src/CommandWithProject.php]: I will add few more commands

File: logging/src/CommandWithProject.php (hunk 5 above)
Comment (tmatsuo) [inline logging/src/CommandWithProject.php]: will do

File: logging/logging-sample-cli.php (hunk 4 above)
Comment (tmatsuo) [inline logging/logging-sample-cli.php]: will do

**Pull request merged**